### PR TITLE
Adds TabBar slider method

### PR DESCRIPTION
**Pull request by wtachau**
@pkluz @allewun

### CODE REVIEW

**Comment by vipulvpatil** [comment]
Does this animate? How? Where?

**Comment by wtachau** [comment]
@vipulvpatil magic

File: Pod/Sources/Categories/UIViewController+FACTabBarSlider.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//  UIViewController+FACTabBarSlider.h
+//  Factiva
+//
+//  Created by William Tachau on 10/1/15.
+//  Copyright © 2015 Originate. All rights reserved.
+//
+
+@import UIKit;
+
+@interface UIViewController (FACTabBarSlider)
```
Comment (pkluz) [inline Pod/Sources/Categories/UIViewController+FACTabBarSlider.h]: Rename category. This is not `FAC` :)

File: Pod/Sources/Categories/UIViewController+FACTabBarSlider.m [hunk 2]
```diff
@@ -0,0 +1,62 @@
+//
+//  UIViewController+FACTabBarSlider.m
+//  Factiva
```
Comment (pkluz) [inline Pod/Sources/Categories/UIViewController+FACTabBarSlider.m]: Change comment names as well.

File: Pod/Sources/Categories/UIViewController+FACTabBarSlider.h (hunk 1 above)
Comment (wtachau) [inline Pod/Sources/Categories/UIViewController+FACTabBarSlider.h]: aha, of course

**Comment by pkluz** [comment]
Okay so let's change the interface of the category to something more generic and flexible. Something that might _just_ have been part of UIKit!

As a user of the category I would expect this type of interaction:
- `setHidingEnabled:(BOOL)enabled forTabBar:(UITabBar *)tabBar whenInteractingWithScrollView:(UIScrollView *)scrollView`.
  - This way you will be able to just pass a pointer to the `tabBarController`'s `tabBar` into the function and not rely on `self.tabBarController` calls.

What do you think? Would such an interface be feasible / possible with your implementation?

**Comment by wtachau** [comment]
Agreed. Making changes now

**Comment by pkluz** [comment]
Update: My previous comment did not the implementation details into account. Suggested adjustment:

```
slideTabBar:(UITabBar)tabBar inResponseToScrollInScrollView:(UIScrollView *)scrollView
```

**Comment by pkluz** [comment]
LGTM

**Pull request closed without merging**